test/shared/src/main/scala-2/zio/test/CompileVariants.scala

```diff
@@ -36,6 +36,7 @@ trait CompileVariants {
    */
   def assertTrue(expr: Boolean, exprs: Boolean*): TestResult = macro SmartAssertMacros.assert_impl
   def assertTrue(expr: Boolean): TestResult = macro SmartAssertMacros.assertOne_impl
+  def assertTrue(exprs: Any*): UIO[Nothing] = macro SmartAssertMacros.assertFlatMapError_impl
```

Comment:
By using `Any`, the 'specialized' boolean overload will be selected unless the ZIO return type is necessary for typechecking.

The function could either return `Nothing` or `TestResult`, but the `Nothing` should help avoid all `test(...)(assertTrue(..))` from being lifted to the ZIO constructor.